src/utils.ts

```diff
@@ -14,3 +17,17 @@ export function deriveBeaconId(airnodeAddress: string, templateId: string) {
 export function deriveBeaconSetId(beaconIds: string[]) {
   return goSync(() => ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['bytes32[]'], [beaconIds]))).data;
 }
+
+export const generateDataFeedBytesSingle = (dataFeed: DataFeedSingle) =>
```

Comment:
This is `deriveBeaconId` function above.

Reply:
Thanks for this, this actually was a mistake @ duplicate.

Reply:
It actually wasn't the same code/implementation - this function encodes a datafeed object as bytes - the whole object, not just the beacon ID.

Reply:
> It actually wasn't the same code/implementation - this function encodes a datafeed object as bytes - the whole object, not just the beacon ID.

🤦  You're right.